feat: include default nunjucks filters and new release to the generator

ayushnau opened the pull request:
Changes Added.

1. Added the new Release with changesets. 

Related issue(s)

Related to: https://github.com/asyncapi/generator/issues/1044

ayushnau commented:
@derberg 
I have added the changes for Changesets.
I have still used the changelog (didnt make it false in the config) cause there is ongoing bug in the changesets which they have not given the solution yet. And this is basically causing changesets still search for the CHANGELOG.md inspite of making it false in the config.
here is the link https://github.com/changesets/action/issues/256

Also I was needed to install the @changesets/cli also cause once I run the chagnesets/cli command directly with npx in ci it is installing the cli version and correctly working fine,   but then the same ci flow changelog is not getting detected causing the error.

derberg commented:
@ayushnau 


1. Regarding your comment about `@changesets/cli` that can be used with npx, but changelong-git must be installed. Please adda proper comment to YAML file, with link to an issue, so we overtime can followup and improve - otherwise this knowledge will get out of our heads in few months.

2. why are there changes to package-lock

3. please add a section in readme, that explain how releasing works now. What has to be done to release minor/patch/major

4. please explain what will happen once we merge this PR. Btw did you test it somewhere already? can you show results?

5. please explore how to make it happen, that new `filters` package is also in dependencies of generator. How to make it happen that once we merge a PR that is marked as "filter", it is automatically bumped in generator

ayushnau commented:
@derberg 
1. Added the comment regarding it. 
2. The Changes in package.lockare expected as the generator-filters is also added. Turborepo will add the required dependency together in package.lock for both the packgaes. Also I have updated the jest version to 27.3.1 without it some test were getting broke. I was getting Error node:events not found. found the solution here https://github.com/Azure/azure-sdk-for-js/issues/29014

3. Added the section in the readmd.md.
4. I thing the repo will be published to respective npm directories will add the test results. 
5. In 5th point you mean. how to bump the version of generator when there is changes in the generator-filter is that right?
If that is the question than it is possible. 
6. Also in beginning you told me to setup the filter in such a way that it is not required to add them to package.json and the filters should get loaded by itself. I checked this one. and found this straightforward. we just need to set the filter object in the     this.templateParams = {};
and update other places if the template params is set to {}.

aeworxet commented:
/bounty

aeworxet commented:
@asyncapi/bounty_team

derberg commented on README.md:
```diff
@@ -1,5 +1,11 @@
 
+This is a Monorepo managed using [Turborepo](https://turbo.build/) and contains the following package:
+
+1. [Generator-filters](apps/generator-filters):This is a library with generator filters that are used in more than one template. Use these filters to not double your work.
```
less important than generator, please move as last on the list and indicate it is only for Nunjucks template and that it is released together with generator by default, that no need to add it to dependencies separately

derberg replied:
and yeah, make sure it is true :) so please add it to generator package.json

derberg commented on README.md:
```diff
@@ -1,5 +1,11 @@
+
+1. [Generator-filters](apps/generator-filters):This is a library with generator filters that are used in more than one template. Use these filters to not double your work.
+
+2. [Generator](apps/generator):This is a tool that you can use to generate whatever you want basing on the AsyncAPI specification file as an input.
```
```suggestion
2. [Generator](apps/generator): This is a tool that you can use to generate whatever you want basing on the AsyncAPI specification file as an input.
```

ayushnau replied:
added the space.

derberg commented on README.md:
```diff
@@ -26,28 +32,54 @@ There is a large number of templates that are ready to use and are officially su
+
+### ---
+
+##### "[package-name-1]": [type] (minor/major/patch)
+
+##### "[package-name-2]": [type]
```
so I can add these 2 in one markdown document, and indicate that one package is fix and another is major and that would work, right?

can you also extend this paragraph with:
- link to some changeset docs
- explanation of the example, so what happens when we merge a PR with such example markdown document - what will show up in npm and what will show up in github releases section

ayushnau replied:
yes we can do that.

derberg replied:
please add this

also something wrong with current formatting of code sample - also notice wrong indentation

![Screenshot 2024-07-22 at 13 51 44](https://github.com/user-attachments/assets/6b427880-06e4-41c6-abca-69282ca3aef9)

derberg reviewed (commented):
damn, I realized that I had some more review questions added in the past....but forgot to submit them and I saw they are still in pending state 🤦🏼 sorry

> I thing the repo will be published to respective npm directories will add the test results.

well before we merge we should be certain and have it tested 😅 `I think` assumption is not enough 😄 

> Also in beginning you told me to setup the filter in such a way that it is not required to add them to package.json and the filters should get loaded by itself. I checked this one. and found this straightforward. we just need to set the filter object in generator this.templateParams = {filter: "@asyncapi/generator-filters"};
and update other places if the template params is set to {}.

but filters are not part of parameters configuration of the template.

this is code responsible for filters registration: https://github.com/asyncapi/generator/blob/master/apps/generator/lib/filtersRegistry.js#L13

as you see it is responsible for calling separate function, one for local filters (the ones defined inside template) and one for config filters (so filters that are added by the name of the package to the config of the generator). You just need to extend it with 3rd line like `addFilters(nunjucks, nunjucks-filters)` with proper comment of course, like why you do it.

`nunjucks-filters` is a name that I suggest (instead of generator-filters, look on other comments), so of course you would have to require it first.

it is very easy to test, you just need to add a sample usage of one of the filters in https://github.com/asyncapi/generator/blob/master/apps/generator/test/test-templates/nunjucks-template/template/test-file.md. I suggest `markdown2html`

might be
```
This is a markdown file for my application.
App name is: **{{ asyncapi.info().title() }}**
Version {{params.version}} running on {{params.mode}} mode

HTML description: **{{ asyncapi.info().description() | markdown2html }}**
```

If you make changes in `filtersRegistry.js` in the right way, and you add a filter usage without adding nunjucks-filters to the config of the template -> the integration tests will start failing - because the snapshot from repo will not match the new one, as the new one will be modified - modified because of filter usage - then just update the snapshot. I hope that makes sense

ayushnau commented on apps/generator-filters/.gitignore:
```diff
@@ -0,0 +1 @@
+node_modules
```
thought we are doing that for generator so lets keep it for generator also though removed now.

ayushnau commented on README.md:
```diff
@@ -26,26 +32,35 @@ There is a large number of templates that are ready to use and are officially su
+
+This library consists of:
+
+- Custom filters. Check out [API docs](docs/api.md) for complete list
```
ok done.

ayushnau commented on package.json:
```diff
@@ -16,7 +16,8 @@
     "generator:lint": "turbo run lint --filter=@asyncapi/generator",
-    "generator:lint:tpl:validator": "turbo run lint:tpl:validator --filter=@asyncapi/generator"
+    "generator:lint:tpl:validator": "turbo run lint:tpl:validator --filter=@asyncapi/generator",
+    "generator-filters:test": "turbo run test --filter=@asyncapi/generator-filters"
```
changed should i change it for package name of generator_filter also to @asyncapi/nunjucks-filters.

derberg replied:
yup - @asyncapi/nunjucks-filters

ayushnau commented on .github/workflows/release-with-changesets.yml:
```diff
@@ -0,0 +1,137 @@
+    branches:
+      - master
+        # The below lines are not enough to have release supported for these branches
+      - next-spec
+      - next-major
+      - next-major-spec
+      - beta
+      - alpha
+      - next
```
you mean changeset here? if yes than I  think we can these branch is supported there is no issue with branch support in changeset.

derberg replied:
changeset - yeah, sorry, still getting used to the name

but as I wrote, these were explicitly supported as they were for pre releases - like for example https://github.com/asyncapi/modelina/releases/tag/v4.0.0-next.54

is changeset supporting such flows?

ayushnau replied:
I think changeset provides the something called `changeset pre` command to releases such branches. I will need to look into it. I am suggesting lets first merge this pr. take this as different issue.

ayushnau commented on .github/workflows/release-with-changesets.yml:
```diff
@@ -0,0 +1,137 @@
+    # Release workflow will be skipped if release conventional commits are not used
+    if: |
+      startsWith( github.repository, 'asyncapi/' ) &&
+      (startsWith( github.event.commits[0].message , 'fix:' ) ||
+      startsWith( github.event.commits[0].message, 'fix!:' ) ||
+      startsWith( github.event.commits[0].message, 'feat:' ) ||
+      startsWith( github.event.commits[0].message, 'feat!:' ))
```
here i also want to discuss this. before answering let me state the flow of changesets. 

1. User create the md file stating the package name and level of change.  and pushes the code. 
2. Now when the ci will run and it will reach to this step:
```js
uses: changesets/action@v1
        id: release
        with:
          version: npx -p @changesets/cli changeset version
          commit: version packages
          title: "feat: version packages"
          publish: npx -p @changesets/cli changeset publish
          setupGitUser: false
```
3. Here the changeset action will go through the md files in .changeset folder and based on that will create the pr. with updated version of package. 
for example in apps/generator
```js 
  "name": "@asyncapi/generator",
  "version": "2.0.1",
```

this will be 

```js
  "name": "@asyncapi/generator",
  "version": "3.0.1",
```
Also this pr will also contain the changelogs that we have listed in the readmd file. (i noticed you were searching in the video you shared there the you need to check the pr that was named chore: version packages.) for example check this pr. https://github.com/ayushnau/parser-js/pull/19

4. Now as the pr is merged (will be done by bot in our case) again the ci will run and once again as the ci reaches to the same changeset action step. it will figure out as the pr is created by it only so this time the publish command will be run and the release will happen.

derberg replied:
ok, found production example -> https://github.com/asyncapi/studio/pull/1112

what I don't understand, why it was triggered by https://github.com/asyncapi/studio/actions/runs/9447823168 if the  PR that triggered this run did not have a changeset file https://github.com/asyncapi/studio/pull/1106/

> Now when the ci will run and it will reach to this step

ok, but this will happen only if conventional commits are respected, fix or feat - which is good

I think mist important change is that no longer a PR with feature triggers release, but a PR created as a result of feature PR - right?

Please provide clear description of that process in readme

ayushnau replied:
https://github.com/asyncapi/studio/actions/runs/9447823168 here I believe (though i didnt understand the complete context) what happening is there was other commits that have the md files like here: https://github.com/asyncapi/studio/commit/2c0ab6cdf18630aebf50d21b933fa0d5d3a16b81 when this ran i think the md that is deleted for that file change might have included already in the previous prs but now as the md is deleted now I believe this is the way for changeset to deal with such problem "what if the md file is deleted later on and the release pr is not released yet" so what it does is it check for previous release and check again after that how many md files are created. and that is how it is figuring out total number of md files.

ayushnau replied:
now for readme you mean to add the release process is that right.

ayushnau replied:
also i checked the documentation there they are suggesting to use the changeset cli to create the md files. 
the documentation whose link https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md

this link needs to be added to the documentation. 

here is how it is working with the use of cli.
[Screencast from 23-07-24 09:49:52 AM IST.webm](https://github.com/user-attachments/assets/b13c43f0-e914-4b95-acca-bbbd0cc3c37a)

ayushnau replied:
please comment on this ???

derberg replied:
we do not need to add dependency, just add instruction how to trigger it with npx
and yeah, if people wanna manual, let them do manual - but instructions lets have CLI examples

ayushnau commented:
@derberg 
test release https://www.npmjs.com/package/@ayushnautiyal706020/generator
first ci link: https://github.com/ayushnau/generator/actions/runs/10017954285
created pr link: https://github.com/ayushnau/generator/pull/6
ci link: https://github.com/ayushnau/generator/actions/runs/10017959613/job/27692765169

ayushnau commented:
@derberg i think i have added all the required functionality. please check.

derberg commented on .changeset/change.md:
```diff
@@ -0,0 +1,5 @@
+---
```
now we have 2 changeset files?

ayushnau replied:
no we have just one. this one needs to deleted.

derberg commented on README.md:
```diff
@@ -26,28 +33,116 @@ There is a large number of templates that are ready to use and are officially su
+
+This library consists of:
+
+- Custom filters. Check out [API docs](apps/generator/docs/api.md) for complete list
```
```suggestion
- Custom filters. Check out [API docs](apps/nunjucks-filters/docs/api.md) for complete list
```

ayushnau replied:
ok updated.

derberg reviewed (commented):
some more docs changes are needed
![Screenshot 2024-07-23 at 12 59 12](https://github.com/user-attachments/assets/442463d7-e098-4c2d-87b7-944295d3403a)
![Screenshot 2024-07-23 at 12 59 30](https://github.com/user-attachments/assets/2d716450-2f8b-4403-8e4f-baf091497037)

derberg commented on .github/workflows/release-with-changesets.yml:
```diff
@@ -0,0 +1,137 @@
+        id: release
+        with:
+          version: npx -p @changesets/cli changeset version
+          commit: "feat: version packages"
+          title: "feat: version packages"
```
```suggestion
          commit: "chore(release): release and bump versions of packages"
          title: "chore(release): release and bump versions of packages"
```

sorry, update also readme pleasae

now since I know how release will work, I understand why initially it was `chore`, lets just add more details

ayushnau replied:
ok done.

derberg commented:
@ayushnau please notice that in my comment about docs there were 2 screenshots about two different docs

derberg commented:
also I see tests are failing, did you change do ker compose setting as mentioned in slack?

ayushnau commented:
> also I see tests are failing, did you change do ker compose setting as mentioned in slack?

i was still working on it. but now it is fixed basically i mounted complete project in the volume. and adjusted other directories used.

ayushnau commented:
@derberg  I think now all tests are passing can we proceed with this now.

ayushnau commented:
> @ayushnau please notice that in my comment about docs there were 2 screenshots about two different docs

checking

derberg commented on .github/workflows/release-with-changesets.yml:
```diff
@@ -19,14 +15,15 @@ on:
       (startsWith( github.event.commits[0].message , 'fix:' ) ||
       startsWith( github.event.commits[0].message, 'fix!:' ) ||
       startsWith( github.event.commits[0].message, 'feat:' ) ||
+      startsWith( github.event.commits[0].message, 'chore(release):' ) ||
```
I don't get why we need it here?

ayushnau replied:
when pr will be created by ci the commit message will be chore(release) if we dont have here the ci will not run. as the commit message is any of fix and feat.

derberg replied:
ok, I'll suggest a change in this workflow - a comment to yaml - that explains it, and you will have a look and commit if I'm right - then we know if we are on the same page

derberg commented on package.json:
```diff
@@ -16,7 +17,9 @@
-    "generator:lint:tpl:validator": "turbo run lint:tpl:validator --filter=@asyncapi/generator"
+    "generator:lint:tpl:validator": "turbo run lint:tpl:validator --filter=@asyncapi/generator",
+    "generator:update:snapshot": "turbo run test:integration:update --filter=@asyncapi/generator",
+    "nunjucks-filters:test": "turbo run test --filter=@asyncapi/generator-filters"
```
is it correct? I think tests will not trigger as `@asyncapi/nunjucks-filters` is what is in the package.json

ayushnau replied:
this is fine this is not for running in ci. when we do npm run test turborepo will go through all the package it have and run the test if they have test script in package.json. this one is only for testing manually. in local.

ayushnau replied:
no this is not correct. fixed it.

derberg commented:
added 2 questions to code and also please have a look on this one that you missed -> https://github.com/asyncapi/generator/pull/1225#issuecomment-2248972858

derberg commented on .changeset/wise-laws-retire.md:
```diff
@@ -0,0 +1,5 @@
+"@asyncapi/generator": minor
+---
+
+Adding new Release Github Flow to the Turborepo.No new features or bugfixes were introduced.
```
```suggestion
- New release pipeline supporting monorepo
- Package `@asyncapi/generator-filters` is now part of `generator` repo and won't be released separately
- By default `@asyncapi/generator-filters` package, known as package with a lot of different nunjucks filters, is registered and added to generator and you no longer have to configure it in your `package.json` 
```

derberg commented on README.md:
```diff
@@ -26,28 +33,122 @@ There is a large number of templates that are ready to use and are officially su
+
 ## Contributing
 
+To Add a major/minor/patch Type Changes:
```
```suggestion
To release a major/minor/patch:
```

derberg commented on README.md:
```diff
@@ -26,28 +33,122 @@ There is a large number of templates that are ready to use and are officially su
+To Add a major/minor/patch Type Changes:
+
+### Manual
+1.  Create a new markdown file in the `.changeset` directory. The filename should be descriptive of the change.
```
```suggestion
1.  Create a new markdown file in the `.changeset` directory. The filename should indicate what the change is about.
```

derberg commented on README.md:
````diff
@@ -26,28 +33,122 @@ There is a large number of templates that are ready to use and are officially su
+1.  Create a new markdown file in the `.changeset` directory. The filename should be descriptive of the change.
+2.  Add the following content to the file in this particular format:
+
+```markdown
+---
+"@package-name-1": [type] (major/minor/patch)
+"@package-name-2": [type]
+---
+
+[Provide a brief description of the changes. For example: Added a new Release GitHub Flow to the Turborepo. No new features or bugfixes were introduced.]
+```
````
```suggestion
    ```markdown
    ---
    "@package-name-1": [type] (major/minor/patch)
    "@package-name-2": [type]
    ---

    [Provide a brief description of the changes. For example: Added a new Release GitHub Flow to the Turborepo. No new features or bugfixes were introduced.]
    ```
```

lmgyuan replied:
After the format example, I would add something like the following to better explain each part of the markdown file:

Inside the file, list the affected packages by their names, such as @package-name-1, followed by the type of change, which can be `major`, `minor`, or `patch`. The type should be specified in parentheses. For example, if you made a non-breaking feature addition to the `@asyncapi/generator` package, you would list it as `"@asyncapi/generator": minor`. 

After listing all relevant packages and their change types, provide a brief description of the changes made. This description should include a clear summary of what was added, fixed, or improved, and why these changes were necessary.

derberg commented on README.md:
````diff
@@ -26,28 +33,122 @@ There is a large number of templates that are ready to use and are officially su
+[Provide a brief description of the changes. For example: Added a new Release GitHub Flow to the Turborepo. No new features or bugfixes were introduced.]
+```
+
+#### For Example:
+```markdown
+---
+"@asyncapi/generator": minor
+---
+
+Adding new Release Github Flow to the Turborepo.No new features or bugfixes were introduced.
+
+```
````
```suggestion
    For Example:
    
    ```markdown
    ---
    "@asyncapi/generator": minor
    ---

    Adding new Release Github Flow to the Turborepo.No new features or bugfixes were introduced.

    ```
```

lmgyuan replied:
After the format example, I would add something like the following to better explain each part of the markdown file:

Inside the file, list the affected packages by their names, such as @package-name-1, followed by the type of change, which can be major, minor, or patch. The type should be specified in parentheses. For example, if you made a non-breaking feature addition to the @asyncapi/generator package, you would list it as "@asyncapi/generator": minor.

After listing all relevant packages and their change types, provide a brief description of the changes made. This description should include a clear summary of what was added, fixed, or improved, and why these changes were necessary.

derberg commented on README.md:
````diff
@@ -26,28 +33,122 @@ There is a large number of templates that are ready to use and are officially su
+
+```
+
+### Using CLI
+```cli 
+npx -p @changesets/cli changeset
+```
+
+3. Simply push the changes with your PR.
````
sorry but what does it mean?

using cli I run npm and then push changes and that is it? where is step 1 and 2? what about the description of the release and the scope? CLI will figure it out on its own?

lmgyuan replied:
I also don't really understand this part. Right now there are three `#` before both Using CLI and Manual, making it seem like "Using CLI" is an alternative to "Manual". However, reading the content, it seems to me that developers should use CLI after they manually add the markdown file in `.changeset`? If my understanding is correct, there should be 4 # before the "Using CLI".

ayushnau replied:
no actually this is alternative to manual.

derberg replied:
@ayushnau remember that if there are 2 folks confused with instructions/code in a similar way, there is no way and you need too address it, improve instructions or add proper code comments for future.

for now I did edit the instructions for you

derberg commented on README.md:
```diff
@@ -26,28 +33,122 @@ There is a large number of templates that are ready to use and are officially su
+3. Simply push the changes with your PR.
+
+
+### Documentation Link 
+For more detailed instructions, you can refer to the official documentation for creating a changeset:
+[Adding a changeset](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md)
```
```suggestion
> [!TIP]
> For more detailed instructions, you can refer to the official documentation for creating a changeset:
[Adding a changeset](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md)
```

derberg commented on README.md:
```diff
@@ -26,28 +33,122 @@ There is a large number of templates that are ready to use and are officially su
+2. **Open a Pull Request**:
+   - Push your changes and open a Pull Request (PR). The changeset file helps communicate the type of changes (major, minor, patch).
+
+3. **CI Processes Changeset**:
```
you mean after PR is merged? please make it clear

ayushnau replied:
done.

derberg commented on README.md:
```diff
@@ -26,28 +33,122 @@ There is a large number of templates that are ready to use and are officially su
+   - Push your changes and open a Pull Request (PR). The changeset file helps communicate the type of changes (major, minor, patch).
+
+3. **CI Processes Changeset**:
+   - During the CI/CD pipeline, the following `changesets/action` step runs:
```
no need to provide content of the workflow, just explanation

ayushnau replied:
done.

derberg commented on README.md:
```diff
@@ -26,28 +33,122 @@ There is a large number of templates that are ready to use and are officially su
+
+   - The new PR will also contain the description from the markdown files.
+
+4. **Merge the Versioning PR**:
```
versioning PR? isn't it also a release triggering PR?

ayushnau replied:
done

derberg commented on README.md:
```diff
@@ -26,28 +33,122 @@ There is a large number of templates that are ready to use and are officially su
+   - Once the versioning PR is created, it should be merged (automatically by the bot in this case).
+
+5. **Release the Package**:
+   - After the PR is merged, the CI/CD pipeline runs again. The `changesets/action` step detects that the PR was created by itself and runs the `publish` command to release the package.
```
can you specify basing on what conditions this happens?

ayushnau replied:
done.

derberg commented on README.md:
```diff
@@ -26,28 +33,122 @@ There is a large number of templates that are ready to use and are officially su
+- Custom filters. Check out [API docs](apps/nunjucks-filters/docs/api.md) for complete list
+- Lodash-powered filters. For the list of all available filters check [official docs](https://lodash.com/docs/)
+
 ## Contributing
```
this section should stay as it was 

all below steps should be in a separate section like `## Release process`

ayushnau replied:
done.

derberg commented on README.md:
```diff
@@ -26,28 +33,122 @@ There is a large number of templates that are ready to use and are officially su
+
 ## Contributing
 
+To Add a major/minor/patch Type Changes:
```
what about conventional commits, they are still used right?

derberg replied:
simply specify that we use conventional commits to have a clear git history of commits and so we can quickly figure out what given commit changed, and if it was release triggering

that feat and fix prefixes are needed to trigger changeset

ayushnau replied:
done.

derberg commented on .github/workflows/release-with-changesets.yml:
```diff
@@ -107,22 +104,28 @@ jobs:
+        run: npm install @changesets/changelog-git@0.2.0
       - if: steps.packagejson.outputs.exists == 'true'
         name: Publish to any of NPM, Github, and Docker Hub
+        uses: changesets/action@v1
```
```suggestion
        #this step has 2 goals, it is either identifying that there is changeset file created and then this action creates a PR with version bump that will trigger release - or if it sees there is no changeset, and there are versions changes in package.json files, it publish new versions to NPM is they are not there yet
        uses: changesets/action@v1
```

lmgyuan commented on README.md:
```diff
@@ -1,5 +1,12 @@
+
+1. [Generator](apps/generator): This is a tool that you can use to generate whatever you want basing on the AsyncAPI specification file as an input.
+
+2. [Generator-filters](apps/nunjucks-filters): This library contains generator filters that can be reused across multiple templates, helping to avoid redundant work. These filters are designed specifically for Nunjucks templates and are included by default with the generator, so there's no need to add then to  dependencies seprately.
```
"no need to add them to  dependencies seprately." instead of "no need to add then to  dependencies seprately."

ayushnau replied:
done.

lmgyuan commented on README.md:
```diff
@@ -1,5 +1,12 @@
+
 ![npm](https://img.shields.io/npm/v/@asyncapi/generator?style=for-the-badge) ![npm](https://img.shields.io/npm/dt/@asyncapi/generator?style=for-the-badge)
 
 > :warning: This package doesn't support AsyncAPI 1.x anymore. We recommend to upgrade to the latest AsyncAPI version using the [AsyncAPI converter](https://github.com/asyncapi/converter-js). If you need to convert documents on the fly, you may use the [Node.js](https://github.com/asyncapi/converter-js) or [Go](https://github.com/asyncapi/converter-go) converters.
```
Delete the first ":" before warning

lmgyuan replied:
This package doesn't support AsyncAPI 1.x anymore. We recommend to upgrade to the latest AsyncAPI version using the [AsyncAPI converter](https://github.com/asyncapi/converter-js) (you can refer to this [installation guide](https://www.asyncapi.com/docs/tools/generator/installation-guide))

ayushnau replied:
done.

lmgyuan commented on README.md:
```diff
@@ -26,28 +33,122 @@ There is a large number of templates that are ready to use and are officially su
 
+# Generator Filters
+
+This library contains generator filters that can be reused across multiple templates, helping to avoid redundant work. These filters are designed specifically for Nunjucks templates and are included by default with the generator, so there's no need to add then to  dependencies seprately.
```
" so there's no need to add them to dependencies seprately."
 instead of 
 " so there's no need to add then to  dependencies seprately."

ayushnau replied:
done.

ayushnau commented:
@derberg @lmgyuan please check made all the required changes.

ayushnau commented:
@derberg @lmgyuan please check the pr.

derberg commented:
/rtm
